baseband/gsb/base.py

```diff
@@ -364,61 +364,29 @@ def __init__(self, fh_ts, fh_raw, sample_rate=None, header=None,
             complex_data=complex_data, samples_per_frame=samples_per_frame,
             payloadsize=payloadsize, sample_rate=sample_rate,
             squeeze=squeeze)
-        self._data = np.zeros((self.samples_per_frame,) + self._unsliced_shape,
-                              (np.complex64 if self.complex_data
-                               else np.float32))
-        self._valid = True
+        self._payload = GSBPayload.fromdata(
+            np.zeros((self.samples_per_frame,) + self._unsliced_shape,
+                     (np.complex64 if self.complex_data else np.float32)),
+            bps=self.bps)
+
+    def _make_frame(self, index):
+        # Set up header for new frame.  (mem_block is set to a rotating
+        # modulo-8 value with no meaning.)
+        time_offset = index * self.samples_per_frame / self.sample_rate
```

Comment:
Why is `self.tell` no longer used in general?  As written, I'm not sure it's possible for something other than `write` to spuriously change the sample position (moreover, you explicitly reset it after each frame write).